clang10 unhappy with Sparta

Clang 10 found two issues in the compilation/testing of Sparta:
```
.././sparta/report/Report.hpp:108:67: error: implicit conversion from 'uint64_t' (aka 'unsigned long') to 'double' changes value from 18446744073709551615 to 18446744073709551616 [-Werror,-Wimplicit-int-float-conversion]
229
            }else if(((uint64_t)val) == val && val >= 0 && val <= ~(uint64_t)0){
230
```
and
```
.././sparta/app/ReportDescriptor.hpp:547:22: note: use reference type 'const std::pair<sparta::Report *, sparta::report::format::BaseFormatter *> &' to prevent copying
239
                for (const auto cur : current) {
240
                     ^~~~~~~~~~~~~~~~
241
                                &
242
```
Mostly easy fixes...

--- a/sparta/sparta/app/ReportDescriptor.hpp
+++ b/sparta/sparta/app/ReportDescriptor.hpp
@@ -544,7 +544,7 @@ namespace sparta {
                 std::vector<Report*> pending = getPendingInstantiations();
 
                 std::unordered_set<Report*> all(pending.begin(), pending.end());
-                for (const auto cur : current) {
+                for (const auto & cur : current) {
                     all.insert(cur.first);
                 }
                 return std::vector<Report*>(all.begin(), all.end());

--- a/sparta/sparta/report/Report.hpp
+++ b/sparta/sparta/report/Report.hpp
@@ -13,7 +13,7 @@
 #include <iomanip>
 #include <sstream>
 #include <unordered_map>
-#include <math.h>
+#include <cmath>
 
 #include "sparta/statistics/StatisticInstance.hpp"
 #include "sparta/kernel/Scheduler.hpp"
@@ -99,18 +99,23 @@ namespace sparta
          */
         static std::string formatNumber(double val,
                                         bool float_scinot_allowed=true,
-                                        int32_t decimal_places=-1) {
+                                        int32_t decimal_places=-1)
+        {
             std::stringstream o;
+            double integral;
+            double fractional = std::modf(val, &integral);
             if(std::isnan(val)){
                 return "nan";
             }else if(std::isinf(val)){
                 o << val; // Use built-in conversion (e.g. +inf, -inf)
-            }else if(((uint64_t)val) == val && val >= 0 && val <= ~(uint64_t)0){
-                // This is an integer. Do not allow scientific notation
-                o << (uint64_t)val;
-            }else if(((int64_t)val) == val && val < 0 && val >= ~(int64_t)0){
-                // This is an integer. Do not allow scientific notation
-                o << static_cast<int64_t>(val);
+            }else if(fractional == 0){
+                if(val < 0) {
+                    // Print as a straight integer, no decimals
+                    o << (int64_t)integral;
+                }
+                else {
+                    o << (uint64_t) integral;
+                }
             }else{
                 if(decimal_places >= 0){
                     o << std::setprecision(static_cast<uint32_t>(decimal_places));
@@ -1689,4 +1694,3 @@ namespace sparta
     }
 
 } // namespace sparta
-

--- a/sparta/test/CMakeLists.txt
+++ b/sparta/test/CMakeLists.txt
@@ -12,6 +12,16 @@ set (VALGRIND_OPTS "--error-exitcode=5 \
       --soname-synonyms=somalloc=NONE")
 set (VALGRIND_TEST_LABEL valgrind_test)
 
+# Since make does not pass the parallel jobs flag to ctest from the user,
+# a fixed count will be set based on core count w/ a max 8
+include(ProcessorCount)
+ProcessorCount(NUM_CORES)
+if (NOT NUM_CORES EQUAL 0)
+  set(CTEST_BUILD_FLAGS -j${NUM_CORES})
+  set(ctest_test_args ${ctest_test_args} PARALLEL_LEVEL ${NUM_CORES})
+endif()
+message(STATUS "Found " ${NUM_CORES} " cores in machine (for ctest)")
+
 # Add the custom regress/regress_valgrind targets.
 add_custom_target (regress)
 add_custom_target (regress_valgrind)
@@ -19,8 +29,8 @@ add_custom_target (regress_valgrind)
 # NOTE:
 # running ctest with --test-action test creates Testing/<datetime>/Test.xml
 # that can be loaded into the CI test result tracker
-add_custom_command (TARGET regress          POST_BUILD COMMAND ctest -LE ${VALGRIND_TEST_LABEL} -j10 --test-action test)
-add_custom_command (TARGET regress_valgrind POST_BUILD COMMAND ctest -L  ${VALGRIND_TEST_LABEL} -j10 --test-action test)
+add_custom_command (TARGET regress          POST_BUILD COMMAND ctest -LE ${VALGRIND_TEST_LABEL} -j${NUM_CORES} --test-action test)
+add_custom_command (TARGET regress_valgrind POST_BUILD COMMAND ctest -L  ${VALGRIND_TEST_LABEL} -j${NUM_CORES} --test-action test)
 
 #add_subdirectory (pipeViewer)
 add_subdirectory (Array)

